hal/src/rtl872x/spi_hal.cpp

```diff
@@ -542,8 +542,12 @@ class Spi {
                 bytesToCopy = std::min(bufferConfig_.rxLength - chunkBuffer_.rxIndex, dmaRxCount + fifoRxCount);
                 if (bufferConfig_.rxBuf) {
                     DCache_Invalidate((u32) chunkBuffer_.rxBuf, chunkBuffer_.rxLength);
+                    int primask = HAL_disable_irq();
+                    __ISB();
```

Comment:
This shouldn't be needed?

Reply:
Only this line?

Reply:
There is no need for DSB/ISB barriers after CPSID. It's not a register write, it's a CPU instruction that is not pipelined.

Reply:
![image](https://user-images.githubusercontent.com/388478/180011705-21fcb370-c195-41fc-b3cc-83661bd45e1a.png)